misc/python/materialize/checks/owners.py

```diff
@@ -42,28 +42,55 @@ def _create_objects(self, role: str, i: int, expensive: bool = False) -> str:
 
         return s
 
-    def _drop_objects(self, i: int) -> str:
-        return dedent(
-            f"""
-            > DROP SECRET owner_secret{i}
-            > DROP MATERIALIZED VIEW owner_mv{i}
-            > DROP VIEW owner_v{i}
-            > DROP INDEX owner_i{i}
-            > DROP TABLE owner_t{i}
-            > DROP TYPE owner_type{i}
-            > DROP CONNECTION owner_csr_conn{i}
-            > DROP CONNECTION owner_kafka_conn{i}
-            > DROP SCHEMA owner_schema{i}
-            > DROP DATABASE owner_db{i}
-            """
+    def _drop_objects(
+        self, role: str, i: int, expensive: bool = False, success: bool = True
+    ) -> str:
+        cmds = [
+            f"DROP SECRET owner_secret{i}",
+            f"DROP MATERIALIZED VIEW owner_mv{i}",
+            f"DROP VIEW owner_v{i}",
+            f"DROP INDEX owner_i{i}",
+            f"DROP TABLE owner_t{i}",
+            f"DROP TYPE owner_type{i}",
+            f"DROP CONNECTION owner_csr_conn{i}",
+            f"DROP CONNECTION owner_kafka_conn{i}",
+            f"DROP SCHEMA owner_schema{i}",
+            f"DROP DATABASE owner_db{i}",
+        ]
+        if expensive:
+            cmds += [
+                f"DROP SOURCE owner_source{i}",
+                f"DROP SINK owner_sink{i}",
+                f"DROP CLUSTER owner_cluster{i}",
+            ]
+        if success:
+            return (
+                f"$ postgres-execute connection=postgres://{role}@materialized:6875/materialize\n"
+                + "\n".join(cmds)
+                + "\n"
+            )
+        if role != "materialize":
+            raise ValueError(
+                "Can't check for failures with user other than materialize"
+            )
+        return "\n".join(
+            [f"! {cmd} CASCADE\ncontains: must be owner of\n" for cmd in cmds]
         )
 
     def _can_run(self) -> bool:
         return self.base_version >= MzVersion.parse("0.47.0-dev")
 
     def initialize(self) -> Testdrive:
         return Testdrive(
-            "> CREATE ROLE owner_role_01"
+            dedent(
+                """
+            $ postgres-execute connection=postgres://mz_system:materialize@materialized:6877
+            ALTER SYSTEM SET enable_rbac_checks TO true
```

Comment:
This is going to affect all Checks that ran after this `initialize()`, right? It may be best to put it somewhere where it will apply to all zippy runs, for example , at line 54 of mzcompose_actions.py and then at `test/cloudtest/test_upgrade.py` in `LiftClusterLimits` .  Those two blocks of `ALTER SYSTEM SET` executions could be unified like this:
- Move LiftClusterLimits from the cloudtest into the platform_checks, rename to `ConfigureInstance` or something.
- Put the `enable_rbac_checks` and `ALTER ROLE materialize` there.
- Make sure that it runs in every Scenario without mentioning it in every scenario. Maybe put it in some list of default Actions that need to run always.
(all of this sounds like a separate PR though).

Reply:
Correct. I thought this might be a good chance to additionally check that things still work with enable_rbac_checks (instead of disabling it again after each step of this check.

Reply:
This ended up quite large so I'll move it to a separate change.

Reply:
I think this block is hard to express in testdrive:
```
        if c.sql_query("SELECT name FROM mz_roles where name = 'materialize'"):
            c.sql(
                "ALTER ROLE materialize CREATEROLE CREATEDB CREATECLUSTER",
                user="mz_system",
                port=6877,
            )
        else:
            c.sql(
                "CREATE ROLE materialize",
                user="mz_system",
                port=6877,
            )
```
Since we don't have `CREATE ROLE IF NOT EXISTS`

Reply:
Yes, the only conditional execution currently supported by testdrive is `skip-if`, that is skip-every-thing-until-the-end-of-the-test-if-this-SQL-query-returns-true . If we decide to keep using testdrive going forward, we may need to eventually invest in some more program flow operators for the testdrive language.

Reply:
Separate change for this: https://github.com/MaterializeInc/materialize/pull/18550